guiEditor/src/components/propertyTab/propertyTabComponent.tsx

```diff
@@ -274,14 +274,27 @@ export class PropertyTabComponent extends React.Component<IPropertyTabComponentP
                         <div id="title">{`${this.state.currentNode.name} [${this.state.currentNode.getClassName()}] (ID: ${this.state.currentNode.uniqueId.toString()})`}</div>
                     </div>
                     {this.renderProperties()}
-                   <hr/>
+                    <hr />
                     <ButtonLineComponent
                         label="REMOVE ELEMENT"
                         onClick={() => {
                             this.state.currentNode?.dispose();
                             this.props.globalState.onSelectionChangedObservable.notifyObservers(null);
                         }}
                     />
+                    <ButtonLineComponent
+                        label="COPY ELEMENT"
+                        onClick={() => {
+                            let serializationObject = {};
+                            this.state.currentNode?.serialize(serializationObject);
+                            let newControl = Control.Parse(serializationObject, this.props.globalState.guiTexture);
+
+                            if (newControl) { //insert the new control into the adt
+                                this.props.globalState.workbench.appendBlock(newControl);

```

Comment:
Should there be any special renaming ???

Reply:
I thought about this but I let @PatrickRyanMS decide what would be best :)

Reply:
I think we should do something to indicate a copied control by adding " copy" to the end of the control name. The user can rename to whatever they want for the final name, but in the interim, it feels like we should delineate any copies as such to prevent confusion about which was the original when looking solely at the tree view. Continued copies should increment like " copy 1", " copy 2" etc. so that they all have unique names. This way, if I make three copies of a button, I can easily identify them to move to a final location and receive a new name.